feat(trace-tree-node): Mitigating SpanNode type guard usage

Abdkhan14 opened the pull request:
- Added a `measurements` getter to the baseNode class and used it to remove SpanNode guards
- Split the common traceSpanRow component that was being used by multiple node classes to their own files

Commit:
feat(trace-tree-node): Mitigating SpanNode type guard usage

Commit:
Merge branch 'abdk/trace-tree-node-missing-ins-guards' into abdk/trace-tree-node-span-node-guards

Commit:
feat(trace-tree-node): Fixing knip errors

gggritso reviewed (commented):
Makes sense, just some questions about code re-use

Abdkhan14 committed:
feat(trace-tree-node): Mitigating eap span node type guards usage (#104497)

- Added an attributes getter to the baseNode class, used it to get rid
of some guards
- Removed isEapTransaction, reduces guards count. guards, we can check
for an eapSpanNode with `value.is_transaction=true`

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>
